tests/core/test_typing.py

```diff
@@ -32,6 +33,12 @@
 logger = logging.getLogger("log")
 
 
+@pytest.fixture(autouse=True)
+def reset__warn_unmapped_properties_log_cache():
+    yield
+    _warn_unmapped_properties.cache_clear()
```

Comment:
This was necessary to get subsequent parametrized runs of `test_conform_object_no_additional_properties` to pass. Maybe there's a better way e.g. make the stream name unique for the parametrized request to avoid a cache hit.

Reply:
> Maybe there's a better way e.g. make the stream name unique for the parametrized request to avoid a cache hit.

That seems like a more elegant solution. Do you think the effort is small enough to do it in this PR? OK if not.

Reply:
https://github.com/meltano/sdk/pull/3109/commits/23322bfe1deb451fe3bc31dcf1ba82e22d1821df